packages/ns-api/files/ns.devices

```diff
@@ -0,0 +1,997 @@
+        case 'active-backup':
+            values = values | {
+                'primary': bond_primary_device,
+                'primary_reselect': 'always',
+                'fail_over_mac': 'none',
+                'num_grat_arp__num_unsol_na': '1'
+            }
+        case 'balance-xor':
+            values = values | {'primary': '', 'xmit_hash_policy': 'layer2'}
+        case 'broadcast':
+            values = values | {'primary': ''}
+        case '802.3ad':
+            values = values | {
+                'min_links': '0',
+                'ad_actor_sys_prio': '65535',
+                'ad_select': 'stable',
+                'lacp_rate': 'slow',
+                'xmit_hash_policy': 'layer2',
+                'primary': ''
+            }
+        case 'balance-tlb':
+            values = values | {
+                'primary': bond_primary_device,
+                'primary_reselect': 'always',
+                'lp_interval': '1',
+                'tlb_dynamic_lb': '1',
+                'xmit_hash_policy': 'layer2'
+            }
+        case 'balance-alb':
+            values = values | {
+                'primary': bond_primary_device,
+                'primary_reselect': 'always',
+                'lp_interval': '1',
+                'xmit_hash_policy': 'layer2',
+                'resend_igmp': '1'
+            }
+
+    return values
+
+
+def set_network_configuration(device_type, interface_name, logical_type, interface_to_edit, protocol, zone, ip4_address, ip4_gateway, ip6_enabled, ip6_address, ip6_gateway, attached_devices, bonding_policy, bond_primary_device, pppoe_username, pppoe_password, dhcp_client_id, dhcp_vendor_class, dhcp_hostname_to_send, dhcp_custom_hostname, uci):
+    if device_type == 'logical' and logical_type == 'bond':
+        values = get_bonding_values(
+            ip4_address, attached_devices, bonding_policy, bond_primary_device)
+    else:
+        # non-bond interfaces
+
+        values = {
+            'proto': protocol,
+            'ipaddr': '',
+            'netmask': '',
+            'ip6addr': '',
+            'gateway': '',
+            'ip6gw': '',
+            'username': '',
+            'password': '',
+            'clientid': '',
+            'vendorid': '',
+            'hostname': '',
+        }
+
+        if protocol == 'static':
+            # ip4 address and netmask
+            values['ipaddr'] = ip4_address
+
+            # ip6 address
+            if ip6_enabled and ip6_address:
+                values['ip6addr'] = ip6_address
+            else:
+                values['ip6addr'] = ''
+
+            if zone == 'wan':
+                # gateway
+                values['gateway'] = ip4_gateway
+
+                # ip6 gateway
+                if ip6_enabled:
+                    values['ip6gw'] = ip6_gateway
+                else:
+                    values['ip6gw'] = ''
+            else:
+                values['gateway'] = ''
+                values['ip6gw'] = ''
```

Comment:
Not sure about this: why gateway is not supported on other zones?

Reply:
Gateway is not displayed on the UI for non-wan zones, so it cannot be set or edited